### DISCUSSION TITLE
Fetching foreign tables

### DISCUSSION THREAD

**Discussion by zidkim** [supabase/supabase#7602] [Questions] [answered]
Hello Supabase Team & Community,

I have two tables: trivia, questions

The schema for both are as follows:
```
trivia
	- id <numeric>
	- date <timestamp>
	- open_door <bool>:false
	- complete <bool>:false
```

```
questions
	- id <numeric>
	- trivia_id <foreign key --- link to trivia.id>
	- question_id <string>
	- question <string>
	- answer <string>
	- choices <array | example - ["one","two",...]>
	- complete <bool>:false
```

I'm trying to fetch a row from the trivia table and the associated questions connected by the foreign key sitting on the questions table (trivia_id).

```
  const supabaseObj = await supabase
    .from("trivia")
    .select(
      `
      *,
      questions (
        complete,
        question,
        choices[],
        trivia
      )
    `
    );

```

The error I get is `column trivia.questions does not exist`. Obviously, I'm doing something wrong but as far as I can tell, I'm simply following docs. That being said, I have little SQL knowledge/experience, so please please feel free to humble me.

What am I doing wrong here?

**Comment by silentworks** [comment]
Can you provide a screenshot of the trivia and questions table inside of the table editor in the Supabase Dashboard, please?

**Reply by zidkim** [reply]
Certainly.

trivia table:
<img width="767" alt="trivia" src="https://user-images.githubusercontent.com/46095969/178058317-b96e0b9c-1576-4360-a73c-d062aa5cdb8b.png">

questions table:
<img width="1419" alt="questions" src="https://user-images.githubusercontent.com/46095969/178058322-72ab4506-ceb7-47f5-af1d-b02905c40df2.png">

**Reply by steve-chavez** [reply]
Is there a foreign key relationship between trivia and questions?

**Reply by zidkim** [reply]
Yes, I've set a foreign key relation on the `questions.trivia` column which points to `trivia.id`.
The goal is to make a schema where each trivia can be associated with multiple questions.

<img width="671" alt="Screen Shot 2022-07-08 at 7 37 41 PM" src="https://user-images.githubusercontent.com/46095969/178088561-4cd740ef-3cc6-4d66-911b-9286b21727ac.png">

**Comment by steve-chavez** [comment, marked as answer]
```js
  const supabaseObj = await supabase
    .from("trivia")
    .select(
      `
      *,
      questions (
        complete,
        question,
        choices[],
        trivia
      )
    `
    );
```

@madleo-dk The `choices[]` above is bad syntax, try just using `choices`.

**Reply by steve-chavez** [reply]
The above should solve it. However we're giving a bad error message in this case, I've opened an issue to track it https://github.com/PostgREST/postgrest/issues/2362.

**Reply by zidkim** [reply]
Thank you @steve-chavez  for pointing that out.
Unfortunately, I'm still not able to pull the questions table -- I'm assuming I've setup this up wrong.
I've replied to @silentworks with some screenshots hoping that may shed some light.

**Reply by steve-chavez** [reply]
> Unfortunately, I'm still not able to pull the questions table -- I'm assuming I've setup this up wrong.

What's the error message?

**Reply by zidkim** [reply]
Ah, sorry I should've stated:
• no errors. 
• fetches the trivia table but nothing from the questions table.

**Reply by steve-chavez** [reply]
Can you share the response output?

**Reply by steve-chavez** [reply]
See https://supabase.com/docs/reference/javascript/select. Make sure to console.log both `data` and `error`.

**Reply by zidkim** [reply]
The issue, aside from the original syntax error, was just a silly mistake in my function. 
Thank you for the help!
